### PR TITLE
feat(jest-environment): add custom matchers

### LINKED ISSUE #83: jest-environment: extend `expect` with custom expectations
We can enable users to have simpler tests by having expectations that match the standard library we provide. For instance, `Response` matchers:

```ts
expect(response).toHaveStatusCode(...);

// These expectations can be shared with `Request` too:
expect(response.headers).toMatchHeaders({ ... });
await expect(response).toHaveJsonBody({ ... });
await expect(response).toHaveTextBody("...");

// or in a single line
await expect(response).toMatchResponse({
  status: ...,
  headers: { ... },
  text: "{ ... }",
  json: {}
});

// streams?
await expect(response.body).toEmitChunk(new Uint8Array([...]));
```

### PR DESCRIPTION
**Pull request by balazsorban44**
Fixes #83

### CODE REVIEW

File: packages/jest-environment/src/extend/response.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+// Matchers
+
+expect.extend({
+  toHaveStatus(actual, ...args: StatusParams) {
+    assertResponse(actual)
+    const [status] = args
+    if (typeof status === 'string') {
+      const ranges = {
```
Comment (Kikobeats) [inline packages/jest-environment/src/extend/response.ts:17]: should `ranges` be declared out of this scope? It looks like it can be reused

File: packages/jest-environment/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/** @type {import("jest").Config} */
+module.exports = {
+  testEnvironment: './dist',
+  setupFilesAfterEnv: ['./jest.setup'],
+  transform: {
+    '\\.(js|jsx|ts|tsx)$': [
+      '@swc/jest',
```
Comment (Kikobeats) [inline packages/jest-environment/jest.config.js:7]: not sure why is this necessary; we're already using ts-jest for transpiling:
https://github.com/vercel/edge-runtime/blob/main/jest.config.ts#L10

is swc solution a faster approach? can we use it in both places?

File: docs/pages/packages/jest-environment.mdx [hunk 3]
```diff
@@ -53,3 +53,61 @@ it('should return the correct value', () => {
   expect(val).toBe(4)
 })
 ```
+
+## Extend expect
+
+The **@edge-runtime/jest-environment** package comes with a sub-module that contains useful [Jest matchers](https://jestjs.io/docs/expect#expectextendmatchers) to test `Request` and `Response`.
```
Comment (Schniz) [inline docs/pages/packages/jest-environment.mdx:59]: Maybe it should be `@edge-runtime/jest-expect` or -matchers? This way people can use it outside the jest env

Comment (balazsorban44) [reply docs/pages/packages/jest-environment.mdx:47]: Sounds good to me!

**Review by Kikobeats — APPROVED** [review]
